src/components/Api.js

```diff
@@ -0,0 +1,90 @@
+class Api {
+  constructor(options) {
+    // constructor body
+    this._baseUrl = options.baseUrl;
+    this._headers = options.headers;
+    this._spinner = document.querySelector(".spinner");
+  }
+
+  _sendRequest(root, options) {
+    return fetch(`${this._baseUrl}${root}`, options)
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+
+        return Promise.reject(`Error: ${res.status}`);
+      })
+      .catch((err) => {
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request and this end is in `index.js` after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.

Reply:
> actually i thought `catch` is always last in chain no matter where in code it is located. because of async.

No, if you add `catch` in the middle of the chain then I can catch the first error **before**  it but the code will proceed after it and can ruit the app